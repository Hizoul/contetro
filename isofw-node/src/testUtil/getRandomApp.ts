import * as authentication from "@feathersjs/authentication"
import * as handler from "@feathersjs/errors/handler"
import * as express from "@feathersjs/express"
import * as rest from "@feathersjs/express/rest"
import feathers, { Application, Service } from "@feathersjs/feathers"
import * as sios from "@feathersjs/socketio"
import * as memdb from "feathers-memory"
import * as mongoService from "feathers-mongodb"
import makeApp from "isofw-node/src/app"
import { DbStore, UserStore } from "isofw-shared/src/util/xpfwdata"
import collections from "isofw-shared/src/xpfwDefs/collections"
import { isString } from "lodash"
import { MongoClient } from "mongodb"
import emptyPort from "./emptyPort"
const sio: any = sios
const res: any = rest
const mongoServic: any = mongoService
const memd: any = memdb

const promisifyListen = (app: any, port: number) => {
  return new Promise((resolve) => {
    const server = app.listen(port, () => {
      resolve(server)
    })
  })
}

const getRandomApp = async (ClientHolder?: any, useRest: boolean = false) => {
  const port = await emptyPort()
  const c = await MongoClient.connect(`mongodb://localhost:27017/`)
  const db  = c.db("contetrotests" + port)
  const app: any = makeApp(undefined, db)
  let server: any
  if (ClientHolder) {
    server = await promisifyListen(app, port)
    await ClientHolder.connectTo(`http://localhost:${port}`, {
      makeAuth: false,
      useRest,
      userStore: UserStore,
      dbStore: DbStore,
      collections
    })
  }
  return {
    app,
    port,
    cleanUp: async () => {
      await db.dropDatabase()
      await c.close()
      if (ClientHolder) {
        ClientHolder.disconnect(useRest)
        server.close()
      }
    }
  }
}

export default getRandomApp
