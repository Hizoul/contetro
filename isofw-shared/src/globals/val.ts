const val = {
  isDebug: true,
  i18nNameSpace: "translations",
  version: "0.5.0",
  service: {
    user: "users",
    play: "play",
    expertData: "expertData",
    gameSession: "gameSession"
  },
  channels: {
    gameUpdates: "gameUpdates"
  },
  animationTime: 1000
}

export default val
