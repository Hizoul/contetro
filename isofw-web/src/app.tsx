import "isofw-shared/src/xpfwDefs"

import url from "isofw-shared/src/globals/url"
import Home from "isofw-web/src/pages/home"
import {
  LoadableAboutPage, LoadableCreateLocalMatch, LoadableCreatePage, LoadableEditPage,
  LoadableExpertOverview, LoadableListPage,
  LoadableLocalGameDetail, LoadableLocalGameHistory,
  LoadableLoginPage,
  LoadableNewOnlineMatch, LoadableOnlineDashboard, LoadableOnlineGameDetail,
  LoadableOnlineMatchHistory, LoadableOnlineMatchOverview, LoadableSettings
} from "isofw-web/src/pages/loadables"
import * as React from "react"
import pose, { PoseGroup } from "react-pose"
import { HashRouter, Link, Route, Switch } from "react-router-dom"
const duration = 8000
const RouteContainer = pose.div({
  enter: {
    opacity: 1, beforeChildren: true, transition: {duration}
  },
  exit: { opacity: 0, transition: {duration}}
})

const AppRouter = () => (
  <HashRouter>
    <Route
      render={({ location }) => (
        <Switch location={location}>
          <Route exact={true} path="/" component={LoadableCreateLocalMatch} key="home" />
          {/* <Route path={url.local} component={LoadableCreateLocalMatch} key={url.local} /> */}
          <Route
            path={`${url.localMatchHistory}/:id`}
            component={LoadableLocalGameHistory}
            key={url.localMatchHistory}
          />
          <Route
            path={`${url.localMatchDetail}/:id`}
            component={LoadableLocalGameDetail}
            key={url.localMatchDetail}
          />
          <Route path="/about" component={LoadableAboutPage} key="about" />
          <Route path={url.settings} component={LoadableSettings} key={url.settings} />
          {/* <Route path="/login" component={LoadableLoginPage} key="login" />
          <Route path={url.newMatch} component={LoadableNewOnlineMatch} key={url.newMatch} />
          <Route
            path={url.onlineMatchOverview}
            component={LoadableOnlineMatchOverview}
            key={url.onlineMatchOverview}
          />
          <Route
            path={`${url.onlineMatchDetail}/:id`}
            component={LoadableOnlineGameDetail}
            key={url.onlineMatchDetail}
          />
          <Route
            path={`${url.onlineMatchHistory}/:id`}
            component={LoadableOnlineMatchHistory}
            key={url.onlineMatchHistory}
          />
          <Route
            path={`${url.expertOverview}`}
            component={LoadableExpertOverview}
            key={url.expertOverview}
          />
          <Route path={url.online} component={LoadableOnlineDashboard} key={url.online} />
          
          <Route path="/create/:collection" component={LoadableCreatePage} key="create" />
          <Route path="/list/:collection" component={LoadableListPage} key="list" />
          <Route path="/edit/:collection/:id" component={LoadableEditPage} key="about" /> */}
        </Switch>
      )}
    />
  </HashRouter>
)

export default AppRouter
