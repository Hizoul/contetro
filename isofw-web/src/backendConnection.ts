import { BackendClient, DbStore, UserStore } from "@xpfw/data"
import { FeathersClient } from "@xpfw/data-feathers"
import url from "isofw-shared/src/globals/url"
import collections from "isofw-shared/src/xpfwDefs/collections"
import { get } from "lodash"

BackendClient.client = FeathersClient

let didConnect = false
const connect = (makeAuth: boolean = true) => {
  if (!didConnect) {
    didConnect = true
    BackendClient.client.connectTo(`${url.webPrefix}${url.mainServer}`, {
      authOptions: {storage: get(global, "window.localStorage")},
      makeAuth,
      useRest: false,
      userStore: UserStore,
      dbStore: DbStore,
      collections
    })
   for (const collection of collections) {
      BackendClient.client.client.service(collection).timeout = 40000
    }
  }
}

export {
  connect
}
