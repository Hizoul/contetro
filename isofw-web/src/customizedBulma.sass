@charset "utf-8";
$topBtnDist: 5pt
#root
  display: block
body, html
  width: 100vw
  height: 100vh
div, body, html
  display: flex
  flex-grow: 1
.inlineBlock
  display: inline-block
  flex-grow: 0
.noFlex
  display: block
  flex-grow: 0
.flex,.flex1
  display: flex
.flex1
  flex: 1
.flexRow
  flex-direction: row !important
.flexColumn
  flex-direction: column !important
.alignCenter
  align-items: center
.justifyCenter
  justify-content: center
.backButton
  position: fixed
  top: $topBtnDist
  left: $topBtnDist
  z-index: 99
  width: 2.5rem
  height: 2.5rem
  opacity: 0.4
  background-color: rgba(0,0,0,0.1)
  display: flex
  justify-content: center
  align-items: center
  color: black
  border-radius: 2rem
  font-size: 2rem
.infoButton
  position: fixed
  top: $topBtnDist
  right: $topBtnDist
  z-index: 99
  width: 2.5rem
  height: 2.5rem
  opacity: 0.4
  background-color: rgba(0,0,0,0.1)
  display: flex
  justify-content: center
  align-items: center
  color: black
  border-radius: 2rem
  font-size: 2rem
.colorContainer
  width: 100%
  div
    display: block
.noPaddingTop
  padding-top: 0pt !important
.makeSquare
  border-radius: 0pt !important
// Set your brand colors
$purple: #4d186e;
$pink: #e91e63;
$brown: #795548;
$beige-light: #4db6ac;
$beige-lighter: #80cbc4;

// Update Bulma's global variables
$primary: #4D186E;
$secondary: #bd2025;
$link: #693987;
$link-invert: white;
$info: #64b5f6;
$info-dark: #1976d2;
// TODO
$success: #64dd17;
$warning: #fdd835;
$danger: #e53935;

$margin: 1rem

$standardTransparency: 0.2;
$transparentPrimary: rgba($primary, $standardTransparency);
$transparentInfo: rgba($info, $standardTransparency);

$pagination-current-background-color: rgba($info, 0.3);
$pagination-current-border-color: rgba($info, 0.3);

$tabs-toggle-link-active-background-color: $primary
$tabs-toggle-link-active-border-color: #6f3cbc

// Import only what you need from Bulma
@import "../node_modules/bulma/sass/utilities/_all.sass";
@import "../node_modules/bulma/sass/base/_all.sass";
@import "../node_modules/bulma/sass/grid/_all.sass";
@import "../node_modules/bulma/sass/components/navbar.sass";
@import "../node_modules/bulma/sass/components/pagination.sass";
@import "../node_modules/bulma/sass/elements/tag.sass";
@import "../node_modules/bulma/sass/elements/table.sass";
@import "../node_modules/bulma/sass/elements/button.sass";
@import "../node_modules/bulma/sass/elements/notification.sass";
@import "../node_modules/bulma/sass/elements/title.sass";
@import "../node_modules/bulma/sass/components/card.sass";
@import "../node_modules/bulma/sass/components/tabs.sass";
@import "../node_modules/bulma/sass/elements/container.sass";
@import "../node_modules/bulma/sass/form/_all.sass";
@import "../node_modules/bulma/sass/layout/hero.sass";

.overlay
  position: fixed
  left: 0px
  top: 0px
  background-color: rgba(0,0,0, 0.6)
  display: flex
  justify-content: center
  align-items: center
  width: 100vw
  height: 100vh
  z-index: 9
  flex-direction: column
  color: white
  text-align: center
  h1
    font-size: 3rem
  h2
    font-size: 2rem
  h3
    font-size: 1.5rem
.divMargin
  div
    margin-bottom: 1rem