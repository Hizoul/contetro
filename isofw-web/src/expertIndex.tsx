import AppRouter from "isofw-web/src/components/expertData/page"
import * as React from "react"
import * as ReactDOM from "react-dom"

document.addEventListener(`DOMContentLoaded`, () => {
  ReactDOM.render(
    (
      <AppRouter />
    ),
    document.getElementById("root")
  )
})
