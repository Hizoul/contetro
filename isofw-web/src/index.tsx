import AppRouter from "isofw-web/src/app"
import * as React from "react"
import * as ReactDOM from "react-dom"

document.addEventListener(`DOMContentLoaded`, () => {
  ReactDOM.render(
    (
      <AppRouter />
    ),
    document.getElementById("root")
  )
})
