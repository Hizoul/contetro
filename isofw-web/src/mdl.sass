@font-face
  font-family: "Roboto"
  src: url("#./Roboto-Regular.ttf") format('truetype')

.grow
  flex-grow: 1

body
  margin: 0pt
  padding: 0pt

.drawerLink
  text-decoration: none

.flex
  display: flex
.flexRow
  flex-direction: row
.flexColumn
  flex-direction: column
.wrap
  flex-wrap: "wrap"
.miniContainer
  width: 98%
  margin-left: auto
  margin-right: auto