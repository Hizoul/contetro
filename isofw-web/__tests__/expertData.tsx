import { BackendClient, DbStore, MailField, PwField, UserStore } from "@xpfw/data"
import FeathersClient from "@xpfw/data-feathers"
import getRandomApp from "isofw-node/src/testUtil/getRandomApp"
import remoteFieldHook from "isofw-shared/src/components/gameFieldProvider/remote"
import val from "isofw-shared/src/globals/val"
import renderSnapshot from "isofw-shared/src/testUtil/renderSnapshot"
import { randomInRange } from "isofw-shared/src/util"
import promiseTimeout from "isofw-shared/src/util/promiseTimeout"
import WebExpertDataCollector from "isofw-web/src/components/expertData"
import * as React from "react"
import useExpertDataCollector from "../../isofw-shared/src/components/expertDataCollect"

BackendClient.client = FeathersClient
const testName = "TESTUSER"
const testPw = "testpw"
test("expertData", async () => {
  const rApp = await getRandomApp(BackendClient.client)
  try {
    await promiseTimeout(400)
    const user = await rApp.app.service(val.service.user).create({
      [String(MailField.title)]: testName,
      [String(PwField.title)]: testPw
    })
    const userId = user._id.toHexString()
    await DbStore.getFromServer(userId, val.service.user)
    renderSnapshot(<WebExpertDataCollector userName={testName} pw={testPw} />,
      "After connected but need login")
    renderSnapshot(<WebExpertDataCollector userName={testName} pw={testPw} />,
      "After connected during login")
    while (UserStore.user == null) {
      await promiseTimeout(400)
    }
    renderSnapshot(<WebExpertDataCollector userName={testName} pw={testPw} />,
      "After connected and logged in")
    console.log("USER IS ", UserStore.user)
    await DbStore.getFromServer("_" + userId, val.service.expertData)
    const expert = useExpertDataCollector(testName, testPw)
    await expert.startPlay()
    const gameId = await DbStore.getFromServer("_" + userId, val.service.expertData)
    await DbStore.getFromServer(gameId, val.service.gameSession)
    renderSnapshot(<WebExpertDataCollector userName={testName} />,
      "After fetched active game")
    let gameOver = false
    console.log("GAME ID IS", gameId)
    let remoteHelper = remoteFieldHook(gameId)
    while (!gameOver) {
      console.log("CALLING PLACE BLOCK")
      await remoteHelper.placeUsingPlay(0)
      remoteHelper = remoteFieldHook(gameId)
      gameOver = remoteHelper.game.gameOver
      // doesn*t end yet
    }
    console.log("ACHIEVED REWARD IS", remoteHelper.game.reward)
    expect(remoteHelper.game.reward !== -1).toBe(true)
  } catch (e) {
    console.log("CAUGHT E", e)
  }
  await rApp.cleanUp()
}, 60000)
