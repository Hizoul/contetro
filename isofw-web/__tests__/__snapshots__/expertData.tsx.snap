// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expertData: After connected and logged in 1`] = `
Array [
  <div
    className="background"
  >
    <div
      className="autoSizer"
    >
       
      <div
        className="gameFieldContainer"
        style={
          Object {
            "minHeight": 200,
            "minWidth": 400,
          }
        }
      >
        <div
          className="line"
          style={
            Object {
              "height": 200,
              "left": 150,
            }
          }
        />
        <div
          className="line"
          style={
            Object {
              "height": 200,
              "left": 250,
            }
          }
        />
        <div
          style={
            Object {
              "backgroundColor": "white",
              "height": 200,
              "left": 150,
              "position": "absolute",
              "width": 100,
            }
          }
        />
      </div>
       
    </div>
  </div>,
  <div
    className="backgroundOverlay"
  >
    <div
      className="backgroundOverlayContent"
    >
      <div>
        Hello 
        TESTUSER
        ! Thanks for willing to participate in my experiment. You can cancel and continue at any time, your progress will be right where you left off :)
        
        <a
          className="overlayButton flex alignCenter blueGradient"
          style={
            Object {
              "width": 146.66666666666666,
            }
          }
        >
          <span
            className="flex alignCenter pullIconLeft"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 640 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M480 96H160C71.6 96 0 167.6 0 256s71.6 160 160 160c44.8 0 85.2-18.4 114.2-48h91.5c29 29.6 69.5 48 114.2 48 88.4 0 160-71.6 160-160S568.4 96 480 96zM256 276c0 6.6-5.4 12-12 12h-52v52c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-52H76c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h52v-52c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h52c6.6 0 12 5.4 12 12v40zm184 68c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48zm80-80c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48z"
              />
            </svg>
          </span>
          <span
            className="flex1 justifyCenter"
          >
            Play
          </span>
        </a>
      </div>
    </div>
  </div>,
]
`;

exports[`expertData: After connected but need login 1`] = `
Array [
  <div
    className="background"
  >
    <div
      className="autoSizer"
    >
       
      <div
        className="gameFieldContainer"
        style={
          Object {
            "minHeight": 200,
            "minWidth": 400,
          }
        }
      >
        <div
          className="line"
          style={
            Object {
              "height": 200,
              "left": 150,
            }
          }
        />
        <div
          className="line"
          style={
            Object {
              "height": 200,
              "left": 250,
            }
          }
        />
        <div
          style={
            Object {
              "backgroundColor": "white",
              "height": 200,
              "left": 150,
              "position": "absolute",
              "width": 100,
            }
          }
        />
      </div>
       
    </div>
  </div>,
  <div
    className="backgroundOverlay"
  >
    <div
      className="backgroundOverlayContent"
    >
      <div>
        You are not logged in yet.
        But the page is still loading so this will probably be done in a jiffy!
      </div>
    </div>
  </div>,
]
`;

exports[`expertData: After connected during login 1`] = `
Array [
  <div
    className="background"
  >
    <div
      className="autoSizer"
    >
       
      <div
        className="gameFieldContainer"
        style={
          Object {
            "minHeight": 200,
            "minWidth": 400,
          }
        }
      >
        <div
          className="line"
          style={
            Object {
              "height": 200,
              "left": 150,
            }
          }
        />
        <div
          className="line"
          style={
            Object {
              "height": 200,
              "left": 250,
            }
          }
        />
        <div
          style={
            Object {
              "backgroundColor": "white",
              "height": 200,
              "left": 150,
              "position": "absolute",
              "width": 100,
            }
          }
        />
      </div>
       
    </div>
  </div>,
  <div
    className="backgroundOverlay"
  >
    <div
      className="backgroundOverlayContent"
    >
      <div>
        You are not logged in yet.
        But the page is still loading so this will probably be done in a jiffy!
      </div>
    </div>
  </div>,
]
`;

exports[`expertData: After fetched active game 1`] = `
<div>
  <div
    className="gameFieldContainer"
    style={
      Object {
        "minHeight": 768,
        "minWidth": 1024,
      }
    }
  >
    <div
      className="line"
      style={
        Object {
          "height": 768,
          "left": 322,
        }
      }
    />
    <div
      className="line"
      style={
        Object {
          "height": 768,
          "left": 702,
        }
      }
    />
    <div
      style={
        Object {
          "backgroundColor": "white",
          "height": 768,
          "left": 322,
          "position": "absolute",
          "width": 380,
        }
      }
    />
    <div
      className="POSEDIV"
    >
      <div
        className="block"
        style={
          Object {
            "backgroundColor": "#000",
            "bottom": 0,
            "left": 0,
            "minHeight": 38,
            "minWidth": 38,
          }
        }
      >
        <div
          className="middleCircle"
          style={
            Object {
              "minHeight": 33,
              "minWidth": 33,
            }
          }
        />
      </div>
      <div
        className="block"
        style={
          Object {
            "backgroundColor": "#000",
            "bottom": 0,
            "left": 38,
            "minHeight": 38,
            "minWidth": 38,
          }
        }
      />
      <div
        className="block"
        style={
          Object {
            "backgroundColor": "#000",
            "bottom": 38,
            "left": 0,
            "minHeight": 38,
            "minWidth": 38,
          }
        }
      />
      <div
        className="block"
        style={
          Object {
            "backgroundColor": "#000",
            "bottom": 38,
            "left": 38,
            "minHeight": 38,
            "minWidth": 38,
          }
        }
      />
    </div>
  </div>
</div>
`;
